Project/src/pages/SpeakingMode/SpeakingMode.ts

```diff
@@ -0,0 +1,189 @@
+
+    constructor(public navCtrl: NavController, public navParams: NavParams, public plt: Platform, private toastCtrl: ToastController, private zone: NgZone) {
+        this.platform = plt;
+
+        this.currUnit = {
+            id: 0,
+            word: '',
+            wordOptions: [],
+            audioPaths: []
+        };
+
+        this.title = navParams.get('sessionTitle');
+        let tempUnits: Promise<screenUnit>[] = navParams.get('screenUnits');
+
+        Promise.all(tempUnits).then((values) => {
+            this.screenUnits = values;
+            this.initUnit();
+        });
+    }
+
+    SpeechToText = function () {
+        this.isCorrect = false;
+        this.platform.ready().then(() => {
+
+            this.recognition = new SpeechRecognition();
+            this.recognition.start();
+            this.recognition.lang = 'en-US';
+            //the voice recognition dont uds what you are speaking -> rarely comes here
+            this.recognition.onnomatch = (event => {
+               this.chooseIncorrect();
+            });
+
+            //either too much noise or something
+            this.recognition.onerror = (event => {
+                
+                this.presentToast("Opps. We didn't hear you correctly, please tap the microphone and pronounce the word again :)")
+            });
+
+            //able to pick up what you are saying
+
+            this.recognition.onresult = event => {
+                this.zone.run(() => { // <== added
+                    var i = 0;
+                    if (event.results.length > 0) {                        
+                        for (i = 0; i < event.results.length; i++) {
+                            //only choose words that are higher than 0.9 confidence
+
+                            //First word is usually the correct word pronounced, but the speech recognition will not be confident sometimes
+                            if (this.currUnit.word === event.results[0][0].transcript.toLowerCase()) {
+                                this.isCorrect = true;
+                                break;
+                            }
+                            //only if confidence level is over 90% for remaining words then it will accept as a right answer
+                            if (event.results[i][0].confidence >= 0.9) {
+                                if (this.currUnit.word === event.results[i][0].transcript.toLowerCase()) {
+                                    this.isCorrect = true;
+                                    break;
+                                }
+                            }
+                        }
+
+                        if (this.isCorrect === true) {
+
+                            this.chooseCorrect();
+                            this.autoAdvance();
+                            return;
+                        }
+						//means out of all the options, non is correct
```

Comment:
Can you fix the weird spacing here? And the extra new lines below? Sorry, I'm OCD... haha...